File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/DeviceEventListFragment.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.text.method.ScrollingMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.hp.workpath.api.Result;
import com.hp.workpath.api.device.events.DeviceEvent;
import com.hp.workpath.sample.deviceeventsample.DeviceEventActivity;
import com.hp.workpath.sample.deviceeventsample.Logger;
import com.hp.workpath.sample.deviceeventsample.MainActivity;
import com.hp.workpath.sample.deviceeventsample.R;
import com.hp.workpath.sample.deviceeventsample.task.DeviceEventTask;
import com.hp.workpath.sample.deviceeventsample.view.DeviceEventListAdapter;

import java.util.List;

import static com.hp.workpath.sample.deviceeventsample.DeviceEventActivity.INDEX;

public class DeviceEventListFragment extends Fragment implements View.OnClickListener {

    private TextView mTotalTextView;
    private TextView mNoEventTextView;
    private TextView mEventTextView;
    private ImageButton mGetDeviceEventListButton;
    private RecyclerView mListView;
    private DeviceEventListAdapter mListAdapter;
    private ProgressBar mProgressBar;

    List<DeviceEvent> mDeviceEvents;
    private static final String SCREEN_4_3_INCH = "Screen_4.3_Inch";


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_device_event_list, container, false);
        if (SCREEN_4_3_INCH.equals(view.findViewById(R.id.layout).getTag())) {
            Toolbar toolBar = view.findViewById(R.id.toolbar);
            ((AppCompatActivity)getActivity()).setSupportActionBar(toolBar);
        }
        mTotalTextView = view.findViewById(R.id.totalTextView);
        mNoEventTextView = view.findViewById(R.id.noEventTextView);
        mEventTextView = view.findViewById(R.id.eventTextView);
        mEventTextView.setMovementMethod(new ScrollingMovementMethod());
        mGetDeviceEventListButton = view.findViewById(R.id.getDeviceEventListButton);
        mGetDeviceEventListButton.setOnClickListener(this);
        mListView = view.findViewById(R.id.deviceEventListView);
        mListAdapter = new DeviceEventListAdapter(listOnClickListener);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        mListView.setLayoutManager(layoutManager);
        mListView.setAdapter(mListAdapter);
        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(requireActivity(),
                DividerItemDecoration.VERTICAL);
        mListView.addItemDecoration(dividerItemDecoration);
        mProgressBar = view.findViewById(R.id.progressBar);
        return view;
    }

    public void getDeviceEventList() {
        mNoEventTextView.setVisibility(View.GONE);
        mProgressBar.setVisibility(View.VISIBLE);
        mTotalTextView.setText("0");
        mListAdapter.clear();
        new DeviceEventTask(getContext(), deviceEventInterface).taskExecute();
    }

    public void setEvent(DeviceEvent deviceEvent) {
        mEventTextView.setText(Logger.build(deviceEvent));
    }

    View.OnClickListener listOnClickListener = new View.OnClickListener() {
        @Override
        public void onClick(View v) {
            int itemPosition = mListView.getChildLayoutPosition(v);
            startDeviceEventActivity(itemPosition + 1);
        }
    };

    private void startDeviceEventActivity(int index) {
        if (index > 0 && index <= mDeviceEvents.size()) {
            Intent intent = new Intent(getContext(), DeviceEventActivity.class);
            intent.putExtra(INDEX, index);
            startActivity(intent);
        } else {
            if (getContext() != null) {
                Logger.showResult(getActivity(), getString(R.string.range_over));
            }
        }
    }

    @Override
    public void onClick(View v) {
        if (v == mGetDeviceEventListButton) {
            getDeviceEventList();
        }
    }

    ResponseInterface deviceEventInterface = new ResponseInterface() {
        @Override
        public void success(List<DeviceEvent> deviceEvents) {
            mProgressBar.setVisibility(View.GONE);
            mDeviceEvents = deviceEvents;
            if (mDeviceEvents != null && mDeviceEvents.size() > 0) {
                mTotalTextView.setText(String.valueOf(mDeviceEvents.size()));
                mListAdapter.setItem(mDeviceEvents);
            } else {
                mNoEventTextView.setVisibility(View.VISIBLE);
            }
        }

        @Override
        public void failure(String msg, Result result) {
            mProgressBar.setVisibility(View.GONE);
            Logger.showResult(getActivity(), msg, result);
        }
    };
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/DeviceEventSample/src/com/hp/workpath/sample/deviceeventsample/fragment/DeviceEventListFragment.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.deviceeventsample.fragment

import android.content.Intent
import android.os.Bundle
import android.text.method.ScrollingMovementMethod
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.hp.workpath.api.Result
import com.hp.workpath.api.device.events.DeviceEvent
import com.hp.workpath.sample.deviceeventsample.DeviceEventActivity
import com.hp.workpath.sample.deviceeventsample.Logger
import com.hp.workpath.sample.deviceeventsample.Logger.build
import com.hp.workpath.sample.deviceeventsample.R
import com.hp.workpath.sample.deviceeventsample.databinding.FragmentDeviceEventListBinding
import com.hp.workpath.sample.deviceeventsample.task.DeviceEventTask
import com.hp.workpath.sample.deviceeventsample.view.DeviceEventListAdapter
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class DeviceEventListFragment : Fragment(), View.OnClickListener {

    private lateinit var mListAdapter: DeviceEventListAdapter
    private lateinit var mDeviceEvents: List<DeviceEvent>
    private var mBindingFragment: FragmentDeviceEventListBinding? = null
    private val mBindingFragmentDeviceEventList get() = mBindingFragment!!
    private val SCREEN_4_3_INCH = "Screen_4.3_Inch"

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mBindingFragment = FragmentDeviceEventListBinding.inflate(inflater, container, false)
        val tag = mBindingFragment?.layout?.tag
        if (SCREEN_4_3_INCH.equals(tag)) {
            (activity as AppCompatActivity).setSupportActionBar(mBindingFragment?.toolbar)
        }
        return mBindingFragmentDeviceEventList.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mBindingFragmentDeviceEventList.eventTextView.movementMethod = ScrollingMovementMethod()
        mBindingFragmentDeviceEventList.getDeviceEventListButton.setOnClickListener(this)
        mListAdapter = DeviceEventListAdapter(listOnClickListener)
        val layoutManager = LinearLayoutManager(context)
        mBindingFragmentDeviceEventList.deviceEventListView.layoutManager = layoutManager
        mBindingFragmentDeviceEventList.deviceEventListView.adapter = mListAdapter
        val dividerItemDecoration = DividerItemDecoration(
            context,
            DividerItemDecoration.VERTICAL
        )
        mBindingFragmentDeviceEventList.deviceEventListView.addItemDecoration(dividerItemDecoration)
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mBindingFragment = null
    }

    fun getDeviceEventList() {
        mBindingFragmentDeviceEventList.noEventTextView.visibility = View.GONE
        mBindingFragmentDeviceEventList.progressBar.visibility = View.VISIBLE
        mBindingFragmentDeviceEventList.totalTextView.text = "0"
        mListAdapter.clear()
        lifecycleScope.launch(Dispatchers.Default) {
            DeviceEventTask(requireContext(), deviceEventInterface).execute()
        }
    }

    fun setEvent(deviceEvent: DeviceEvent?) {
        mBindingFragmentDeviceEventList.eventTextView.text = deviceEvent?.let { build(it) }
    }

    private var listOnClickListener = View.OnClickListener { v ->
        val itemPosition =
            mBindingFragmentDeviceEventList.deviceEventListView.getChildLayoutPosition(v)
        startDeviceEventActivity(itemPosition + 1)
    }

    private fun startDeviceEventActivity(index: Int) {
        if (index > 0 && index <= mDeviceEvents.size) {
            val intent = Intent(context, DeviceEventActivity::class.java)
            intent.putExtra(DeviceEventActivity.INDEX, index)
            startActivity(intent)
        } else {
            Logger.showResult(requireActivity(), getString(R.string.range_over))
        }
    }

    override fun onClick(v: View) {
        when (v) {
            mBindingFragmentDeviceEventList.getDeviceEventListButton -> {
                getDeviceEventList()
            }
        }
    }

    private var deviceEventInterface: ResponseInterface = object : ResponseInterface {
        override fun success(deviceEvents: List<DeviceEvent>?) {
            mBindingFragmentDeviceEventList.progressBar.visibility = View.GONE
            mDeviceEvents = deviceEvents as List<DeviceEvent>
            if (mDeviceEvents.isNotEmpty()) {
                mBindingFragmentDeviceEventList.totalTextView.text = mDeviceEvents.size.toString()
                mListAdapter.setItem(mDeviceEvents as MutableList<DeviceEvent>)
            } else {
                mBindingFragmentDeviceEventList.noEventTextView.visibility = View.VISIBLE
            }
        }

        override fun failure(msg: String?, result: Result?) {
            mBindingFragmentDeviceEventList.progressBar.visibility = View.GONE
            Logger.showResult(activity, msg, result)
        }
    }

}
